docs/tutorials/handler.md

```diff
@@ -0,0 +1,389 @@
+
+The editor we ran to add our event handler to our project provided all
+the boilerplate for our handler, so we just need to focus on the
+event-handling logic in the `#!typescript handle` method.  Because of
+the third argument `#!typescript @EventHandler` annotation,
+`#!typescript "/Issue()"`, this handler will only be fired when events
+about issues occur.  Inside the `#!typescript handle` method, we have
+added a conditional on the state of the evented issue such that if the
+issue is not in a closed state, our handler will return an empty plan,
+i.e., essentially a no-op.  If the issue is closed, we construct a
+thank you message for each of the assignees and add it to the plan.
+
+With the code in place, let's try to run the tests again.
+
+<div class="ss-container">
+  <img src="../images/rug-passing-tests.png" alt="TDD Rug passing tests" class="ss-large">
+</div>
+
+Success!  Now it's time to publish test your new handler, making it
+available in chat for everyone on your team.
+
+## Publish your handler
+
+To make Atomist aware of your new handler you need to publish the
+handler project.  This is often done via continuous integration but
+for our purposes here you're going to see how it's done manually using
+the Rug CLI.
+
+### Configure the Rug CLI
+
+To publish new Rugs you need to configure the Rug CLI with the
+credentials it needs to push your Rug project archive to Atomist.  The
+first step in configuring credentials for publishing is to log in to
+Atomist using the `rug repositories login` command.  You will be
+prompted for your GitHub user credentials.
+
+```console
+$ rug repositories login
+
+
+The Rug CLI needs your GitHub login to identify you.
+
+The command will create a GitHub Personal Access Token with scope 'read:org'
+which you can revoke any time on https://github.com/settings/tokens.  Your
+password will not be displayed or stored. Your sensitive information will not
+be sent to Atomist; only to api.github.com.
+
+  → Username : .....
+  → Password : .....
+
+  Please provide a MFA code
+  → MFA code : .....
+
+Successfully logged in to GitHub and stored token in ~/.atomist/cli.yml
+```
+
+!!! important ""
+    No passwords are stored, only a unique personal access token with
+    read:org scope.
+
+!!! note "Two-Factor Authentication"
+    If your GitHub account is configured to use two-factor
+    authentication, you will be prompted for the MFA code as shown
+    above.
+
+Once you are logged in, you can run the `rug repositories configure`
+to save your publishing configuration to your Rug CLI configuration
+file.
+
+```console
+$ rug repositories configure
+
+→ Repositories
+  <unique identifier of your team's repository>
+    <your unique team's repository url here>
+```
+
+You're now all set to publish your handlers project.
+
+### Publishing
+
+You use the Rug CLI `publish` command to publish a Rug project archive
+to Atomist.  You execute the `rug publish` command from the handlers
+project directory:
```

Comment:
`...command from the handlers project directory` is a little unclear.
Perhaps `..command from the root of the handlers project directory`?